apps/emqx/test/emqx_topic_SUITE.erl

```diff
@@ -128,6 +129,21 @@ t_match_perf(_) ->
     true = match(Name, Filter),
     ok = bench('match/2', fun emqx_topic:match/2, [Name, Filter]).
 
+t_intersect(_) ->
+    <<"t/global/1/+">> = intersection(<<"t/global/#">>, <<"t/+/1/+">>),
+    <<"t/global/#">> = intersection(<<"t/global/#">>, <<"#">>),
+    <<"t/global/#">> = intersection(<<"t/global/#">>, <<"t/global/#">>),
+    <<"1/2/3/4/5">> = intersection(<<"1/+/3/+/5/#">>, <<"+/2/+/4/+">>),
+    <<"t/local/1">> = intersection(<<"t/local/1/#">>, <<"t/local/+">>),
+    false = intersection(<<"t/global/#">>, <<"t/local/+">>),
+    false = intersection(<<"t/local/1/+">>, <<"t/local/+">>).
+
+t_sys_intersect(_) ->
+    <<"$SYS/broker/+">> = intersection(<<"$SYS/broker/#">>, <<"$SYS/+/+">>),
+    <<"$SYS/broker">> = intersection(<<"$SYS/broker">>, <<"$SYS/+">>),
+    false = intersection(<<"$SYS/broker">>, <<"+/+">>),
+    false = intersection(<<"$SYS/broker">>, <<"#">>).
+
```

Comment:
Maybe it's worth adding a few tests to intersect wildcard with non-wildcard and two non-wildcards? Since the function is expected to work with any valid topic/filter pair..
Something like:
```
<<"t/test/1">> = intersection(<<"t/test/#">>,<<"t/test/1">>),
<<"t/test/1/1">> = intersection(<<"t/test/1/1">>, <<"t/test/#">>),
false = intersection(<<"t/test/1/1">>, <<"t/test/+">>),
<<"t/test/1/1">> = intersection(<<"t/test/1/1">>, <<"t/test/1/1">>),
false = intersection(<<"t/test/1">>, <<"t/test/2">>),
false = intersection(<<"t/test/1">>, <<"t/test/1/2">>)
```